internal/compiler/generator/cpp.rs

```diff
@@ -951,16 +952,11 @@ fn generate_item_tree(
     target_struct.members.push((
         Access::Private,
         Declaration::Function(Function {
-            name: "parent_item".into(),
-            signature: "(slint::private_api::ComponentRef component, uintptr_t index, slint::private_api::ItemWeak *result) -> void".into(),
+            name: "parent_node".into(),
+            signature: "([[maybe_unused]] slint::private_api::ComponentRef component, [[maybe_unused]] slint::private_api::ItemWeak *result) -> void".into(),
             is_static: true,
             statements: Some(vec![
-                format!("auto self = reinterpret_cast<const {}*>(component.instance);", item_tree_class_name),
-                "if (index == 0) {".into(),
                 parent_item_from_parent_component,
```

Comment:
This could actually be a vec  since it may even be empty in the `else` case and could then contain two lines.

Reply:
Yeap, that is probably better. I'll change that.